docker: Use devel package instead of spelling out duplicate dependencies

okurz opened the pull request:
Follow-up to https://github.com/os-autoinst/openQA/pull/2295 . After #2295 is in and the package openQA-devel is available we can try to build the container by switching from devel:openQA:Leap:42.3 to devel:openQA.

Testing in https://build.opensuse.org/package/show/home:okurz:branches:devel:openQA:feature:use_openqa_devel/openqa_dev

okurz commented:
Follow-up to https://github.com/os-autoinst/openQA/pull/2295 . After #2295 is in and the package openQA-devel is available we can try to build the container by switching from devel:openQA:Leap:42.3 to devel:openQA

coolo commented on openQA.spec:
```diff
@@ -45,15 +45,20 @@
-# sub-packages except for the devel package
-%define test_requires %common_requires %main_requires %python_scripts_requires %worker_requires perl(Test::MockModule) perl(Test::Output)
-%define devel_requires %test_requires
+%define main_requires %assetpack_requires git-core perl(Carp::Always) perl(Date::Format) perl(DateTime::Format::Pg) perl(DBD::Pg) perl(DBI) >= 1.632, perl(DBIx::Class) => 0.082801, perl(DBIx::Class::DeploymentHandler) perl(DBIx::Class::DynamicDefault) perl(DBIx::Class::Schema::Config) perl(DBIx::Class::Storage::Statistics) perl(DBIx::Class::OptimisticLocking) perl(File::Copy::Recursive) perl(Net: ...
```
I can't say that having all in one line is any readable. Just stop using macros, *especially* if only used once.

okurz replied:
that's actually a change that was already merged as part of #2295 . I am investigating how to make that more readable and at best also share the list with cpanfile but I consider using the openQA-devel package in the container definition more important and then to make it possible to define new dependencies *and* test as part of the same PR.

coolo replied:
what you should do IMO:

Create another spec file openQA-devel.spec that build requires nothing, but requires the things you need to require. There can also be subpackages for the things you put in macros now. This spec file then hardcodes some dependencies like git-core and chromedriver and takes the rest out of cpanfile by spec file magic.

You can run shell scripts to generate requires in a lot of different ways - e.g. how OBS used to generate its ruby dependencies:
https://build.opensuse.org/package/view_file/OBS:Server:2.6/obs-server/obs-api-deps.spec?expand=1 (line 41)

coolo commented:
can you rebase?

okurz commented:
rebased. There are two changes, both only in the Dockerfile used on OBS, switching the repo from the Leap sub-repo to the parent repo `devel:openQA` so that we have openQA-devel and os-autoinst-devel available as well as trying to install openQA-devel instead of the spelled out dependency list.

Unfortunately the build of the container in https://build.opensuse.org/package/show/home:okurz:branches:devel:openQA:feature:use_openqa_devel/openqa_dev still fails. I now added the repo path "<path project="devel:openQA:Leap:42.3" repository="openSUSE_Leap_42.3"/>" which again yields a failure to find the container base template. I could again explicitly mention the base template project as it's apparently not inherited from the intermediate repo or we just merge as is and check in devel:openQA itself if dependencies resolve properly.
